pom.xml

```diff
@@ -678,6 +680,14 @@
                     <groupId>org.wildfly.plugins</groupId>
                     <artifactId>wildfly-maven-plugin</artifactId>
                     <version>${version.wildfly.plugin}</version>
+                    <!-- Need to override the plugin dependency to make jboss-logging lib work -->
+                    <dependencies>
+                        <dependency>
+                            <groupId>org.jboss.logging</groupId>
+                            <artifactId>jboss-logging</artifactId>
+                            <version>${version.org.jboss.logging.jboss-logging}</version>
+                        </dependency>
+                    </dependencies>
```

Comment:
Is there some kind of error if we don't override this? That feels more like a bug we should look at if that's the case.

Reply:
yup... if we don't override it there's:
```java
[ERROR] Failed to execute goal org.wildfly.plugins:wildfly-maven-plugin:5.0.0.Final:provision (server-provisioning) on project wildfly-core-testsuite-embedded: Provisioning failed: Forked embedded process has failed: 'java.lang.Object org.jboss.logging.Logger.getMessageLogger(java.lang.invoke.MethodHandles$Lookup, java.lang.Class, java.lang.String)' -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.wildfly.plugins:wildfly-maven-plugin:5.0.0.Final:provision (server-provisioning) on project wildfly-core-testsuite-embedded: Provisioning failed
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:50)
Caused by: org.apache.maven.plugin.MojoExecutionException: Provisioning failed
    at org.wildfly.plugin.provision.AbstractProvisionServerMojo.execute (AbstractProvisionServerMojo.java:274)
    at org.wildfly.plugin.provision.ProvisionServerMojo.execute (ProvisionServerMojo.java:21)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:50)
Caused by: org.jboss.galleon.ProvisioningException: Forked embedded process has failed
    at org.wildfly.galleon.plugin.server.ForkedEmbeddedUtil.fork (ForkedEmbeddedUtil.java:152)
    at org.wildfly.galleon.plugin.server.ForkedEmbeddedUtil.fork (ForkedEmbeddedUtil.java:84)
    at org.wildfly.galleon.plugin.config.generator.WfConfigGenerator.doGenerate (WfConfigGenerator.java:174)
    at org.wildfly.galleon.plugin.config.generator.WfConfigGenerator.generate (WfConfigGenerator.java:122)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.wildfly.galleon.plugin.WfInstallPlugin.invokeConfigGenerator (WfInstallPlugin.java:1235)
    at org.wildfly.galleon.plugin.WfInstallPlugin.generateConfigs (WfInstallPlugin.java:805)
    at org.wildfly.galleon.plugin.WfInstallPlugin.postInstall (WfInstallPlugin.java:455)
    at org.jboss.galleon.runtime.ProvisioningRuntime$3.visitPlugin (ProvisioningRuntime.java:337)
    at org.jboss.galleon.runtime.ProvisioningRuntime$3.visitPlugin (ProvisioningRuntime.java:334)
    at org.jboss.galleon.layout.ProvisioningLayout$Handle.visitPlugins (ProvisioningLayout.java:290)
    at org.jboss.galleon.layout.ProvisioningLayout.visitPlugins (ProvisioningLayout.java:863)
    at org.jboss.galleon.runtime.ProvisioningRuntime.provision (ProvisioningRuntime.java:334)
    at org.jboss.galleon.ProvisioningManager.doProvision (ProvisioningManager.java:650)
    at org.jboss.galleon.ProvisioningManager.provision (ProvisioningManager.java:409)
    at org.jboss.galleon.caller.ProvisioningContextImpl.provision (ProvisioningContextImpl.java:110)
    at org.jboss.galleon.api.ProvisioningImpl.provision (ProvisioningImpl.java:279)
    at org.jboss.galleon.api.Provisioning.provision (Provisioning.java:130)
    at org.wildfly.plugin.provision.AbstractProvisionServerMojo.provisionServer (AbstractProvisionServerMojo.java:313)
    at org.wildfly.plugin.provision.AbstractProvisionServerMojo.execute (AbstractProvisionServerMojo.java:266)
    at org.wildfly.plugin.provision.ProvisionServerMojo.execute (ProvisionServerMojo.java:21)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:50)
Caused by: java.lang.NoSuchMethodError: 'java.lang.Object org.jboss.logging.Logger.getMessageLogger(java.lang.invoke.MethodHandles$Lookup, java.lang.Class, java.lang.String)'
    at org.wildfly.core.embedded.logging.EmbeddedLogger.<clinit> (EmbeddedLogger.java:33)
    at org.wildfly.core.embedded.Configuration$Builder.setModulePath (Configuration.java:230)
    at org.wildfly.core.embedded.EmbeddedProcessFactory.createHostController (EmbeddedProcessFactory.java:203)
    at org.wildfly.galleon.plugin.config.generator.BaseConfigGenerator.doStartHc (BaseConfigGenerator.java:172)
    at org.wildfly.galleon.plugin.config.generator.ForkedConfigGenerator.executeScript (ForkedConfigGenerator.java:77)
    at org.wildfly.galleon.plugin.config.generator.ForkedConfigGenerator.forkedForEmbedded (ForkedConfigGenerator.java:45)
    at org.wildfly.galleon.plugin.server.ForkedProcessRunner.main (ForkedProcessRunner.java:56)
```

Reply:
> It might be as simple as upgrading jboss-logging in the plugin.

+1 I gave it a try and it worked, https://github.com/wildfly/wildfly-maven-plugin/blob/5981abafe1ecaf8f4caddaf24ccf864eee9f60d9/pom.xml#L75 updating to 3.6.0 allows to remove this dependency override

Reply:
https://github.com/wildfly/wildfly-maven-plugin/pull/548

Reply:
@marko-bekhta @jamezp , a new release of WildFly Galleon Plugins 7.1.2.Final fixes the logging dependency leaking at provisioning time.

Reply:
nice! 😃 I'll give it a try in this PR. Thanks for letting us know.

Reply:
@marko-bekhta Can you rebase on main and remove this dependency?

Reply:
Hey, yes sure. Let me get back to my workstation first 🙈 I'll try to update it later today or tomorrow morning 😊

Reply:
done 😃 
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  12:36 min
[INFO] Finished at: 2024-07-30T21:19:17+02:00
[INFO] ------------------------------------------------------------------------
```
seems to be working fine, at least locally 🙈 🎉